altair/utils/data.py

```diff
@@ -80,41 +80,37 @@ def sample(data, n=None, frac=None):
 
 
 @curry
-def to_json(data, prefix='altair-data'):
-    """Write the data model to a .json file and return a url based data model."""
-    check_data_type(data)
-    ext = '.json'
-    filename = _compute_filename(prefix=prefix, ext=ext)
-    if isinstance(data, pd.DataFrame):
-        data = sanitize_dataframe(data)
-        data.to_json(filename, orient='records')
-    elif isinstance(data, dict):
-        if 'values' not in data:
-            raise KeyError('values expected in data dict, but not present.')
-        values = data['values']
-        with open(filename) as f:
-            json.dump(values, f)
+def to_json(data, prefix='altair-data', extension='json',
+            filename="{prefix}-{hash}.{extension}"):
+    """
+    Write the data model to a .json file and return a url based data model.
+    """
+    data_json = _data_to_json_string(data)
```

Comment:
I tested the logic locally and it works as expected. In particular I made multiple plots of the same dataset (one file) and then mutated the data frame (second file). Very nice!